java/src/main/java/org/rocksdb/RocksDB.java

```diff
@@ -307,7 +309,9 @@ public static RocksDB open(final DBOptions options, final String path,
     db.storeOptionsInstance(options);
 
     for (int i = 1; i < handles.length; i++) {
-      columnFamilyHandles.add(new ColumnFamilyHandle(db, handles[i]));
+      final ColumnFamilyHandle columnFamilyHandle = new ColumnFamilyHandle(db, handles[i]);
+      db.ownedColumnFamilyHandles.add(columnFamilyHandle);
+      columnFamilyHandles.add(columnFamilyHandle);
     }
 
```

Comment:
Instead of `columnFamilyHandles.add(columnFamilyHandle);` within the loop, you could just do `columnFamilyHandles.addAll(db.ownedColumnFamilyHandles)` after the loop. A bulk add should be more efficient than adding one by one.

This could also be done in the other open methods.